### PR TITLE
Fix for identity-3171 to populate AttributeServiceIndex correctly

### PR DESCRIPTION
**Pull request by ChamaraPhilipsuom**
Fix for Identity-3171 issue. "authnReqDTO" now contains AttributeConsumingServiceIndex read from SAML Request.

### CODE REVIEW

File: components/sso-saml/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/admin/SAMLSSOConfigAdmin.java [hunk 1]
```diff
@@ -96,8 +96,13 @@ public boolean addRelyingPartyServiceProvider(SAMLSSOServiceProviderDTO serviceP
             } else {
                 serviceProviderDO.setAttributeConsumingServiceIndex(Integer.toString(IdentityUtil.getRandomInteger()));
             }
+            serviceProviderDO.setEnableAttributesByDefault(serviceProviderDTO.isEnableAttributesByDefault());
         } else {
             serviceProviderDO.setAttributeConsumingServiceIndex("");
+            if (serviceProviderDO.isEnableAttributesByDefault()) {
+                log.warn("Enable Attribute Profile must be selected to activate it by default. EnableAttributesByDefault will be disabled.");
```
Comment (omindu) [inline components/sso-saml/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/admin/SAMLSSOConfigAdmin.java]: Check line width. Seems too long.

Comment (ChamaraPhilipsuom) [inline components/sso-saml/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/admin/SAMLSSOConfigAdmin.java]: Fixed at https://github.com/wso2/carbon-identity/pull/1682

File: components/sso-saml/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/util/SAMLSSOUtil.java [hunk 2]
```diff
@@ -930,25 +930,35 @@ public static boolean validateXMLSignature(RequestAbstractType request, String a
...
 
             if (request.getAttributeConsumingServiceIndex() == null) {
-                if (authnReqDTO.getAttributeConsumingServiceIndex() != 0) {
-                    index = authnReqDTO.getAttributeConsumingServiceIndex();
-                    spDO.setAttributeConsumingServiceIndex(String.valueOf(index));
+                //SP has not provide a AttributeConsumingServiceIndex in the authnReqDTO
+                if (StringUtils.isNotBlank(spDO.getAttributeConsumingServiceIndex())) {
+                    if (spDO.isEnableAttributesByDefault()) {
+                        index = Integer.parseInt(spDO.getAttributeConsumingServiceIndex());
+                    } else {
```
Comment (pulasthi7) [inline components/sso-saml/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/util/SAMLSSOUtil.java]: can get rid of this else block by combining the two if conditions at L940,941 with &&.

File: components/sso-saml/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/util/SAMLSSOUtil.java [hunk 3]
```diff
@@ -930,25 +930,35 @@ public static boolean validateXMLSignature(RequestAbstractType request, String a
...
                 index = request.getAttributeConsumingServiceIndex();
             }
         } else {
-            index = authnReqDTO.getAttributeConsumingServiceIndex();
-            if (index != 0) {
-                spDO.setAttributeConsumingServiceIndex(String.valueOf(index));
+            if (StringUtils.isNotBlank(spDO.getAttributeConsumingServiceIndex())) {
+                if (spDO.isEnableAttributesByDefault()) {
+                    index = Integer.parseInt(spDO.getAttributeConsumingServiceIndex());
+                } else {
```
Comment (pulasthi7) [inline components/sso-saml/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/util/SAMLSSOUtil.java]: can be removed as mentioned in above comment

File: components/sso-saml/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/util/SAMLSSOUtil.java (hunk 2 above)
Comment (ChamaraPhilipsuom) [inline components/sso-saml/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/util/SAMLSSOUtil.java]: Fixed at #1682

File: components/sso-saml/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/util/SAMLSSOUtil.java (hunk 3 above)
Comment (ChamaraPhilipsuom) [inline components/sso-saml/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/util/SAMLSSOUtil.java]: Fixed at #1682

**Pull request merged**